core/providers/cohere.go

```
@@ -418,6 +462,30 @@ func (provider *CohereProvider) ChatCompletion(ctx context.Context, model, key s
 	return bifrostResponse, nil
 }
 
+// processImageContent processes image content for Cohere API format.
+// It creates a copy of the image content, normalizes and formats it, then returns the properly formatted map.
+// This prevents unintended mutations to the original image content.
+func processImageContent(imageContent *schemas.ImageContent) map[string]interface{} {
```

golangci-lint (1.64.8)

func `processImageContent` is unused

(unused)